devito/at_controller.py

```diff
@@ -0,0 +1,344 @@
+            ValueError: if isat installation directory does not exist
+        """
+        if not os.path.exists(isat_dir_path):
+            raise ValueError("isat installation directory does not exits")
+
+        self.time_order = None
+        self.space_order = None
+
+        self._isat_py_path = os.path.join(isat_dir_path, "Source/Python/isat.py")  # path to isat script that does at
+        self._at_work_dir = os.path.join(isat_dir_path, "Auto-tuning")  # working dir for at
+
+        at_file_dir = os.path.join(self._at_work_dir, "Isat-files")  # dir for keeping files necessary for at
+
+        # all these files are necessary for auto tuning process and they have to be in same dir as tuned file
+        # do not change these names
+        self._isat_build_f = "isat-build"
+        self._isat_clean_f = "isat-clean"
+        self._isat_test_f = "isat-test"
+        self._make_f = "makefile"
+
+        self._build_f_path = os.path.join(at_file_dir, self._isat_build_f)
+        self._clean_f_path = os.path.join(at_file_dir, self._isat_clean_f)
+        self._test_f_path = os.path.join(at_file_dir, self._isat_test_f)
+        self._make_f_path = os.path.join(at_file_dir, self._make_f)
+
+        self._at_parent_folder = ""  # saves the location of current at parent folder
+        self._at_src_dir = "src"  # source folder where file which we'll auto tune will be kept
+        self._at_dst_dir = "dst"  # destination folder where at result will be kept
+
+        # for simplicity renaming all the tuned files to this name
+        # their parent dir will contain the time and space order in cpp file kernel
+        self._at_f_name = "tune_me"
+        shell_ref_str = "#!/bin/sh"
+
+        if not os.path.exists(self._at_work_dir):  # creates main working directory
+            os.mkdir(self._at_work_dir)
+
+        # Uncomment if contents of isat files changes
+        # shutil.rmtree(at_file_dir)
+
+        if not os.path.exists(at_file_dir):  # creates directory where all necessary files for at will be kept
+            os.mkdir(at_file_dir)
+
+            with open(self._build_f_path, 'w') as f:  # creates isat-build
+                f.writelines([shell_ref_str, "\nmake build"])
+            self._set_x_permission(self._build_f_path)
+
+            with open(self._clean_f_path, 'w') as f:  # create isat-clean
+                f.writelines([shell_ref_str, "\nmake clean"])
+            self._set_x_permission(self._clean_f_path)
+
+            with open(self._test_f_path, 'w') as f:  # create isat-test
+                f.writelines([shell_ref_str, "\n./%s.exe" % self._at_f_name])
+            self._set_x_permission(self._test_f_path)
+
+            with open(self._make_f_path, 'w') as f:  # creates make-file
+                # copied make file contents from one of examples in isat folder. If there's a need feel free to change
+                make_file_contents = ["CXX = icc", "\nCXX_OPTS = -O3 -qopenmp",  # compiler and its options
```

Comment:
Gonna extract compiler from DEVITO_CC however didn't you say that youre planning removing all env vars in the future? About -qopenmp I though it just tells compiler to look at the pragmas? so it doesn't mater if we running without openmp flag or with, as compiler wont find any pragmas.